### PR TITLE
Generating extension service assembly definitions & folder clean up

### PR DESCRIPTION
**Pull request by radicalad**
## Overview
This PR adds the proper assembly definitions for the existing MRTK extension services.
- New assembly definitions for the Extensions `SceneTransitionService` and `LostTrackingService`, including `editor` where appropriate.
- Added folder structure to match `UnityAR` data provider extension service.

## Changes
- Fixes: #6562

### CODE REVIEW

File: Assets/MixedRealityToolkit.Extensions/SceneTransitionService/Interfaces.meta [hunk 1]
```diff
@@ -0,0 +1,8 @@
+fileFormatVersion: 2
```
Comment (davidkline-ms) [inline Assets/MixedRealityToolkit.Extensions/SceneTransitionService/Interfaces.meta]: nit: this file isnt needed as there is no Editor folder.

File: Assets/Build.meta [hunk 2]
```diff
@@ -1,5 +1,5 @@
 fileFormatVersion: 2
```
Comment (keveleigh) [inline Assets/Build.meta]: What's this folder meta? `Assets/Build.meta`

Comment (radicalad) [reply Assets/Build.meta]: No idea. I do a `git clean -df` regularly, but Unity still manages to generate weird files. I'll remove 😄

**Pull request merged**